include/mockturtle/views/cnf_view.hpp

```diff
@@ -109,6 +121,20 @@ class cnf_view : public Ntk
    */
   inline std::optional<bool> solve( std::vector<int> assumptions, int limit = 0 )
   {
+    if ( ps_.write_dimacs )
+    {
+      for ( auto l : assumptions )
+      {
+        dimacs_.add_clause( &l, &l + 1 );
+      }
+
+      dimacs_.set_nr_vars( solver_.nr_vars() );
+
+      auto fd = fopen( ps_.write_dimacs->c_str(), "w" );
```

Comment:
Just a quick question: why do you use `fopen` over `std::ofstream`? Either way, you probably want to check if the stream is actually opened before writing to it, don't you?

Reply:
Hi, thanks for the comment. The *percy* API only offers `FILE*`. However, I am currently rewriting this part using [bill](https://github.com/lsils/bill) and hope it'll be much cleaner then.

Reply:
Thanks for taking the time to provide your insight!